### PR TITLE
[TD-12715]<fix>(connector):fix get json metadata

### PR DESCRIPTION
**Pull request by huolibo**

### CODE REVIEW

File: src/connector/jdbc/src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java [hunk 1]
```diff
@@ -232,6 +234,7 @@ public boolean getBoolean(int col) throws SQLException {
             }
 
             case TSDBConstants.TSDB_DATA_TYPE_NCHAR:
+                case TSDBConstants.TSDB_DATA_TYPE_JSON:
```
Comment (sangshuduo) [inline src/connector/jdbc/src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java]: indent issue

File: src/connector/jdbc/src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java [hunk 2]
```diff
@@ -271,6 +274,7 @@ public long getLong(int col) throws SQLException {
...
```
Comment (sangshuduo) [inline src/connector/jdbc/src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java]: indent issue

File: src/connector/jdbc/src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java [hunk 3]
```diff
@@ -308,6 +312,7 @@ public double getDouble(int col) {
...
```
Comment (sangshuduo) [inline src/connector/jdbc/src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java]: indent issue

File: src/connector/jdbc/src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java (hunk 1 above)
Comment (huolibo) [reply src/connector/jdbc/src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java]: just for CI. not finish

**Review by dingbo8128 — APPROVED** [review]
LGTM

**Review by zyyang-taosdata — APPROVED** [review]
LGTM

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**